src/layers/mbconv.jl

```diff
@@ -0,0 +1,117 @@
+                       bias::NTuple{2, Bool} = (!use_norm[1], !use_norm[2]), kwargs...)
+    return vcat(conv_norm(kernel_size, inplanes, inplanes, activation; eps,
+                          revnorm, use_norm = use_norm[1], stride, bias = bias[1],
+                          groups = inplanes, kwargs...),
+                conv_norm((1, 1), inplanes, outplanes, activation; eps,
+                          revnorm, use_norm = use_norm[2], bias = bias[2]))
+end
+
+# TODO add support for stochastic depth to mbconv and fused_mbconv
+"""
+    mbconv(kernel_size, inplanes::Integer, explanes::Integer,
+                     outplanes::Integer, activation = relu; stride::Integer,
+                     reduction::Union{Nothing, Integer} = nothing)
+
+Create a basic inverted residual block for MobileNet variants
+([reference](https://arxiv.org/abs/1905.02244)).
+
+# Arguments
+
+  - `kernel_size`: kernel size of the convolutional layers
+  - `inplanes`: number of input feature maps
+  - `explanes`: The number of feature maps in the hidden layer
+  - `outplanes`: The number of output feature maps
+  - `activation`: The activation function for the first two convolution layer
+  - `stride`: The stride of the convolutional kernel, has to be either 1 or 2
+  - `reduction`: The reduction factor for the number of hidden feature maps
+    in a squeeze and excite layer (see [`squeeze_excite`](#))
+"""
+function mbconv(kernel_size::Dims{2}, inplanes::Integer, explanes::Integer,
+                outplanes::Integer, activation = relu; stride::Integer,
+                dilation::Integer = 1, reduction::Union{Nothing, Integer} = nothing,
+                norm_layer = BatchNorm, momentum::Union{Nothing, Number} = nothing,
+                se_from_explanes::Bool = false, divisor::Integer = 8, no_skip::Bool = false)
+    @assert stride in [1, 2] "`stride` has to be 1 or 2 for `mbconv`"
+    # handle momentum for BatchNorm
+    if !isnothing(momentum)
+        @assert norm_layer==BatchNorm "`momentum` is only supported for `BatchNorm`"
+        norm_layer = (args...; kwargs...) -> BatchNorm(args...; momentum, kwargs...)
+    end
+    layers = []
+    # expand
+    if inplanes != explanes
+        append!(layers,
+                conv_norm((1, 1), inplanes, explanes, activation; norm_layer))
+    end
+    # depthwise
+    append!(layers,
+            conv_norm(kernel_size, explanes, explanes, activation; norm_layer,
+                      stride, dilation, pad = SamePad(), groups = explanes))
+    # squeeze-excite layer
+    if !isnothing(reduction)
+        squeeze_planes = _round_channels((se_from_explanes ? explanes : inplanes) ÷
+                                         reduction, divisor)
+        push!(layers,
+              squeeze_excite(explanes, squeeze_planes; activation, gate_activation = hardσ))
+    end
+    # project
+    append!(layers, conv_norm((1, 1), explanes, outplanes, identity))
+    use_skip = stride == 1 && inplanes == outplanes && !no_skip
```

Comment:
Is `no_skip` used somewhere?

Reply:
For completeness. We toggle `no_skip` deliberately to `false` inside the builders so that the main CNN builder can add the `Parallel`, but for someone not expecting this behaviour they should be able to use the layer independently and get the `SkipConnections` as intended when the stride is  1 and `inplanes == outplanes`

Reply:
But that stride and channel logic isn't part of the paper "theory," right? It's a heuristic for automated building which isn't being done when the layer is used independently. It seems less magical to me for someone who wants to use this independently to explicitly write:
```julia
SkipConnection(mbconv(...), +)
```
And we avoid more keywords switches.

Reply:
> But that stride and channel logic isn't part of the paper "theory," right?

Umm...this is kinda debatable, I guess? MobileNetv2 (which introduces the inverted residual block) talks about the residual connection in the inverted residual block as being key to the design. In fact I was planning on adding these same toggles to the ResNet blocks for completeness since the way I see it a residual block which doesn't return with a residual connection doesn't feel right. We can always toggle this in relevant builders (you're right that it is one more keyword to handle, though)

Reply:
There are many papers on ResNet style blocks and even more code implementations. So terminology gets all mixed up. But in the original paper, "residual" specifically refers to a single branch of the skip connection. If we want to learn a mapping $H(x)$, then the block learns $H(x) = F(x) + x$ (in the basic identity block) where $F(x) = H(x) - x$ is the "residual," i.e. what _remains_ when $x$ is taken out of $H(x)$.

On a more practical note, I'd prefer to avoid keyword switches for a non-essential task that is trivial to do manually. It's always possible to add these features later if someone wants them, but no point complicating an already complex function prematurely.